lib/Cache/RedisDB.pm

```diff
@@ -149,7 +153,8 @@ Return a list of all known keys in the provided I<$namespace>.
 sub keys {
     my ($self, $namespace) = @_;
     my $prefix = _cache_key($namespace, undef);
-    return [map { s/^$prefix//; $_ } @{redis->keys($prefix . '*')}];
+    my $pl = length($prefix);
+    return [map { substr($_, $pl) } @{redis->keys($prefix . '*')}];
```

Comment:
Yes.  Large key lists (we have several hundred thousand in some namespaces) see a real difference with this change.